### PR TITLE
Script injector component test

### PR DESCRIPTION
**Pull request by ShivanKaul**
Add dedicated browsertests for new Script Injector component.

Resolves https://github.com/brave/brave-browser/issues/34167

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

These are browsertests.

### CODE REVIEW

File: browser/script_injector/script_injector_browsertest.cc [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  net::EmbeddedTestServer https_server_;
+
+ private:
+  BraveContentBrowserClient client_;
+  content::ContentMockCertVerifier mock_cert_verifier_;
+};
+
+// TESTS
+
+IN_PROC_BROWSER_TEST_F(ScriptInjectorBrowserTest, ScriptInjected) {
```
Comment (bridiver) [inline browser/script_injector/script_injector_browsertest.cc:108]: ScriptInjector can be used without awaiting a promise so maybe this should be `InjectScriptAwaitPromise` or something and then similar for the others?

File: browser/script_injector/script_injector_browsertest.cc [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+      blink::mojom::UserActivationOption::kDoNotActivate,
+      blink::mojom::PromiseResultOption::kAwait, std::move(callback));
+  // Wait until the callback is called.
+  run_loop.Run();
+  std::u16string expected_title(u"test");
+  content::TitleWatcher watcher(web_contents(), expected_title);
+  EXPECT_EQ(expected_title, watcher.WaitAndGetTitle());
+}
+
+IN_PROC_BROWSER_TEST_F(ScriptInjectorBrowserTest, ScriptInjectedReturnsFalse) {
```
Comment (bridiver) [inline browser/script_injector/script_injector_browsertest.cc:129]: I'm not sure this test adds a lot of value. You just happen to have chosen a bool in the previous test so I think the correct result is sufficient. I think it would be more useful to test passing an empty callback? I think you could use `base::MockCallback` and mock `is_null` to return `true` and then check to make sure it never gets called?

Comment (ShivanKaul) [reply browser/script_injector/script_injector_browsertest.cc]: Added a new test for a different return type (dictionary), and unit tests for the `CheckIfWantResult` method.

File: components/script_injector/renderer/BUILD.gn [hunk 3]
```diff
@@ -20,3 +20,14 @@ component("renderer") {
...
+
+source_set("unit_tests") {
+  testonly = true
+  sources = [ "script_injector_unittest.cc" ]
+  deps = [
+    "//base/test:test_support",
+    "//brave/components/script_injector/renderer:renderer",
+    "//components/prefs:test_support",
+  ]
+  defines = [ "HAS_OUT_OF_PROC_TEST_RUNNER" ]
```
Comment (bridiver) [inline components/script_injector/renderer/BUILD.gn:32]: this is only needed for browser tests https://source.chromium.org/chromium/chromium/src/+/main:content/public/test/browser_test.h;l=49?q=browser_test.h&ss=chromium

File: components/de_amp/browser/test/BUILD.gn [hunk 4]
```diff
@@ -13,7 +13,6 @@ source_set("unit_tests") {
     "//base/test:test_support",
     "//components/prefs:test_support",
   ]
-  defines = [ "HAS_OUT_OF_PROC_TEST_RUNNER" ]
```
Comment (ShivanKaul) [inline components/de_amp/browser/test/BUILD.gn:16]: Don't need this either.

**Review by bridiver — APPROVED** [review]
